File: bioinformatics_methods_2/module08/parse_blast.py
#!/usr/bin/env python3
# Clint Valentine
# 3/2/2016

import re

blast_output = '/scratch/RNASeq/blastp.outfmt6'
parsed_blast = './parsed_blast.txt'

# Open BLAST output (reading) and an outfile (writing) for parsed data.
with open(blast_output) as infile, open(parsed_blast, 'w') as outfile:
    for line in infile:
        # Split on occurences of either a tab or pipe from the stripped line.
        line = re.split(r'[\t\|]', line.strip())

        # Unpack the line and save the required fields.
        transcript_id, isoform, _, _, _, swiss_id, _, identity, *_ = line

        # Write the fields tab-delimited to the end of outfile.
        line = '\t'.join([transcript_id, isoform, swiss_id, identity]) + '\n'
        outfile.write(line)

File: bioinformatics_methods_2/module02/parseBlast.pl
#!/usr/bin/perl
# Clint Valentine
# 1/27/2016
use warnings;
use strict;

my $infile = "/scratch/RNASeq/blastp.outfmt6";

parseBlast();

sub parseBlast {
	# Open filehandle for blast output
	open( BLAST6, "<", $infile ) or die $!;

	# Read entry line-by-line
	while (<BLAST6>) {
		chomp;

		my $pattern = qr/
			(?<transcript>.*?)\|   # Match transcript field
			(?<isoform>.*?)\tgi\|  # Match isoform field
			(?<gi>.*?)\|sp\|       # Match GI ID field
			(?<sp>.*?)\|           # Match sp field
			(?<prot>.*?)\t         # Match protien ID field
			(?<pident>.*?)\t       # Match percent identity field
			(?<length>.*?)\t       # Match length field
			(?<mismatch>.*?)\t     # Match mismatch field
			(?<gapopen>.*?)\t      # Match gaps open field
			/x;

		# Match line to pattern and print out groups in tab delimination
		if ( $_ =~ /$pattern/ ) {
			print join "\t", $+{transcript}, $+{isoform}, $+{gi}, $+{sp}, $+{prot},
				$+{pident}, $+{length}, $+{mismatch}, $+{gapopen} . "\n";
		}
	}
}
